resources/asciidoctor/lib/elastic_compat_preprocessor/extension.rb

```diff
@@ -119,77 +119,115 @@ class ElasticCompatPreprocessor < Asciidoctor::Extensions::Preprocessor
   LEGACY_INLINE_MACRO_RX = /(#{LEGACY_MACROS})\[([^\]]*)\]/
 
   def process(_document, reader)
-    reader.instance_variable_set :@in_attribute_only_block, false
-    reader.instance_variable_set :@code_block_start, nil
-    def reader.process_line(line)
+    reader.extend ReaderExtension
+  end
+
+  ##
+  # Extensions to the Reader object that implement the conversions.
+  module ReaderExtension
```

Comment:
I feel like this makes it much more readable, mostly because I can break things into named methods.